libmcount/mcount.c

```diff
@@ -1356,7 +1356,7 @@ static int __mcount_entry(unsigned long *parent_loc, unsigned long child, struct
 		return -1;
 	}
 
-	if (unlikely(mtdp->in_exception)) {
+	if (!mcount_estimate_return && unlikely(mtdp->in_exception)) {
```

Comment:
Thanks for the fix.  I think it's better to move this check to `mcount_rstack_reset_exception()` to minimize the change and to be in sync with `mcount_rstack_reset()`.

Reply:
Ok, I also found that if I don't do this, `mtdp->in_exception` will not be reset to false, which may cause other issues.
Similarly, I removed the check for `mcount_estimate_return` in `__cxa_begin_catch()`.

Reply:
Actually I don't remember why I put it in `__cxa_begin_catch()` but it seems ok.

Reply:
This line is introduced by [27b0f44](https://github.com/namhyung/uftrace/commit/27b0f44ef2fae114a3b14bedcfd71e4861269abf)